### PR TITLE
BUG: Series.map is ignoring the na_action keyword

### LINKED ISSUE #46588: BUG: `Series.map` is ignoring the `na_action` keyword
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd
a = pd.Series([pd.NA, 1, 1, pd.NA])
a.map({1:2, pd.NA:3}, na_action=True)
```

### Issue Description

This code should throw an error because `na_action` is invalid, but instead I've gotten a Jupyter notebook, ipython in Powershell, and an online interpreter to generate the following output
```
0    3
1    2
2    2
3    3
dtype: int64
```
I get the same output for `na_action='ignore'` and `na_action=None`

### Expected Behavior

`ValueError: na_action must either be 'ignore' or None, True was passed`

### Installed Versions

<details>
INSTALLED VERSIONS
------------------
commit           : 06d230151e6f18fdb8139d09abf539867a8cd481
python           : 3.9.12.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.22000
machine          : AMD64
processor        : Intel64 Family 6 Model 141 Stepping 1, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_United States.1252

pandas           : 1.4.1
numpy            : 1.22.3
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 22.0.4
setuptools       : 58.1.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : 7.30.1
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : 1.4.28
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None

</details>

### PR DESCRIPTION
**Pull request by stanleycai95**
- [x] closes #46588 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest doc/source/whatsnew/v1.5.0.rst file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/base.py [hunk 1]
```diff
@@ -842,6 +842,13 @@ def _map_values(self, mapper, na_action=None):
                 )
 
         if isinstance(mapper, ABCSeries):
+            # na_action must be None or "ignore"
+            if na_action not in (None, "ignore"):
```
Comment (mroeschke) [inline pandas/core/base.py:846]: A test is needed to exercise this `if` branch

**Review by mroeschke — CHANGES_REQUESTED** [review]
Also good to have a whatsnew entry in `1.5.0.rst` (probably in the Other Enhancement section as a improvement in error reporting)

**Comment by stanleycai95** [comment]
Added documentation updates + test. Thank you for the feedback!!

File: pandas/core/base.py (hunk 1 above)
Comment (stanleycai95) [reply pandas/core/base.py:846]: Added a test, thank you!

File: pandas/tests/apply/test_invalid_arg.py [hunk 2]
```diff
@@ -66,6 +66,15 @@ def test_map_with_invalid_na_action_raises():
         s.map(lambda x: x, na_action="____")
 
 
+def test_map_arg_is_dict_with_invalid_na_action_raises():
+    # https://github.com/pandas-dev/pandas/issues/46588
+    s = Series([1, 2, 3])
+    na_action = "____"
```
Comment (mroeschke) [inline pandas/tests/apply/test_invalid_arg.py:72]: Could you use `pytest.mark.parameterize` to test `"____"` and `True` (like in the original issue)?

Comment (stanleycai95) [reply pandas/tests/apply/test_invalid_arg.py]: Yeah definitely, it's done now.

File: doc/source/whatsnew/v1.5.0.rst [hunk 3]
```diff
@@ -79,6 +79,7 @@ as seen in the following example.
 
 Other enhancements
 ^^^^^^^^^^^^^^^^^^
+- :meth:`Series.map` now raises when ``arg`` is dict but ``na_action`` is not either None or 'ignore' (:issue:`46588`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.5.0.rst:82]: can you use double back-ticks around ``None`` and ``'ignore'``

File: pandas/core/base.py [hunk 4]
```diff
@@ -842,6 +842,13 @@ def _map_values(self, mapper, na_action=None):
                 )
 
         if isinstance(mapper, ABCSeries):
+            # na_action must be None or "ignore"
```
Comment (jreback) [inline pandas/core/base.py:845]: comment not necessary

**Review by jreback — CHANGES_REQUESTED** [review]
minor comment, pls merge master and ping on green

File: doc/source/whatsnew/v1.5.0.rst (hunk 3 above)
Comment (stanleycai95) [reply doc/source/whatsnew/v1.5.0.rst]: Yeah definitely! Done.

File: pandas/core/base.py (hunk 4 above)
Comment (stanleycai95) [reply pandas/core/base.py]: Thanks! Removed.

**Review by stanleycai95 — COMMENTED** [review]
Thank you so much!! Made corresponding changes.

**Review by mroeschke — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @stanleycai95 very nice!

**Pull request merged**